src/Agent.Sdk/Util/ParentProcessUtil/WindowsParentProcessUtil.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Agent.Sdk.Util.ParentProcessUtil
+{
+    public enum ParentProcessNames { Pwsh, Powershell };
+
+    public static class WindowsParentProcessUtil
+    {
+        public static (bool, Dictionary<string, string>) IsParentProcess(bool useInteropToFindParentProcess, params ParentProcessNames[] processNames)
+        {
+            var processList = new List<Process>();
+
+            try
+            {
+                (processList, var telemetry) = GetProcessList(Process.GetCurrentProcess(), useInteropToFindParentProcess);
+
+                bool isProcessRunningInPowerShell = processList.Exists(process => IsProcessPowershell(process, processNames));
+
+                return (isProcessRunningInPowerShell, telemetry);
+            }
+            finally
+            {
+                foreach (var process in processList)
+                {
+                    process.Dispose();
```

Comment:
Better to not create a list of processes as that keeps unmanaged resource around.  Just store the info you need from the process (e.g. process name / id) and dispose process as you go